Data/Aeson/Types/Generic.hs

```diff
@@ -211,11 +210,18 @@ class TaggedObject' f isRecord where
     taggedObjectPairs' :: Options -> String -> f a -> Tagged isRecord [Pair]
     taggedObjectEnc' :: Options -> String -> f a -> Tagged isRecord B.Builder
 
+instance OVERLAPPING_ TaggedObject' U1 True where
```

Comment:
Hi @devwout, I think it would be better not to treat U1 as a record. I think the following is simpler.

Encoding:

```haskell
instance OVERLAPPING_ TaggedObject' U1 False where
    taggedObjectPairs' _opts _contentsFieldName =
        Tagged . const []
    {-# INLINE taggedObjectPairs' #-}

    taggedObjectEnc' _opts _contentsFieldName =
        Tagged . const mempty
    {-# INLINE taggedObjectEnc' #-}
```

Decoding:

```haskell
instance OVERLAPPING_ FromTaggedObject'' U1 False where
    parseFromTaggedObject'' _opts _contentsFieldName = Tagged . const (pure U1)
    {-# INLINE parseFromTaggedObject'' #-}
```

Could you try if that works and update your PR?